src/externs.ts

```diff
@@ -543,21 +564,32 @@ export function generateExterns(
    * Adds an import alias for the symbol defined at the given node. Creates an alias name based on
    * the given moduleName and (optionally) the name.
    */
-  function addImportAlias(node: ts.Node, moduleName: string, name: ts.Identifier|string|undefined) {
+  function addImportAlias(
+      node: ts.Node, moduleName: string, isGoogImport: boolean,
+      name: ts.Identifier|string|undefined) {
     let symbol = typeChecker.getSymbolAtLocation(node);
     if (!symbol) {
       reportDiagnostic(diagnostics, node, `named import has no symbol`);
       return;
     }
+    if (symbol.flags & ts.SymbolFlags.Alias) {
+      symbol = typeChecker.getAliasedSymbol(symbol);
+    }
+    // While type_translator does add the mangled prefix for ambient declarations, it only does so
+    // for non-aliased (i.e. not imported) symbols. That's correct for its use in regular modules,
+    // which will have a local symbol for the imported ambient symbol. However within an externs
+    // file, there are no imports, so we need to make sure the alias already contains the correct
+    // module name, which means the mangled module name in case of imports symbols.
+    // This only applies to non-Closure ('goog:') imports.
```

Comment:
does that fix my clutz import/export issue? I don't see any test with 'goog:' changing.

Reply:
Also, I don't undestand what id non-Closure (goog:) import, isn't 'goog:' only closure imports?

Reply:
A Closure import is an import with `'goog:'` prefix. A non-Closure import is any other import. This code only applies to the non-Closure bit.

I think this is independent of your import/export issue (though I might be misunderstanding that). There's a test for this, which is (paraphrapsing) `import X from 'goog:y.X'; export declare const x: X;` in a `.d.ts` file. The important bit is that we emit the `y.X` type as `y.X` even in the externs file.